src/unity/lib/annotation/image_classification.cpp

```diff
@@ -0,0 +1,287 @@
+#include "image_classification.hpp"
+#include <functional>
+
+#include <unity/lib/gl_sarray.hpp>
+
+namespace turi {
+namespace annotate {
+
+ImageClassification::ImageClassification(
+    const std::shared_ptr<unity_sframe> &data,
+    const std::vector<std::string> &data_columns,
+    const std::string &annotation_column)
+    : AnnotationBase(data, data_columns, annotation_column) {
+  /* Since abstract classes cannot use std::make_shared and it needed for the
+   * annotate process, we need the child class to construct the shared pointer
+   * and store it in the protected variables of the inherited abstract
+   * `AnnotationBase` class */
+  m_self = std::make_shared<ImageClassification>(*this);
```

Comment:
The make_shared(*this) pattern is pretty weird. `this` is then also going to be holding a shared pointer to a copy of `this`.

Instead, I would let this class behave normally (just store its own instance members), and make the caller of this constructor use make_shared instead.

Reply:
Fixed in ` [eb82ffc](/apple/turicreate/pull/1423/commits/eb82ffc4e5ae687f048637a5ec5899a62f57acaa)`